core/src/main/java/org/fao/geonet/kernel/security/openidconnect/OIDCConfiguration.java

```diff
@@ -0,0 +1,220 @@
+     */
+    public String idTokenRoleLocation = "groups";
+
+    /**
+     * Converts roles from the OIOC server to GN profiles (or in the form of "GN-group:GN-profile").
+     * You can specify via setRoleConverterString in the form:
+     * "OIDCServerRole1=GNProfile,OIDCServerRole2=GROUP:PROFILE"
+     */
+    public Map<String, String> roleConverter = new HashMap<>();
+
+    /**
+     * All users who login via the OIDC will have this profile (at a minimum).
+     * This is useful to allow ALL users in an org to login to GN without having to setup all users.
+     * Typically, this should be GUEST or REGISTEREDUSER.  But, you can make it higher (i.e. Editor).
+     */
+    public String minimumProfile = "Guest";
+
+    /**
+     * if true, always update the GN user with information from OIDC.
+     */
+    public boolean userProfileUpdateEnabled = true;
+
+    /**
+     * if true, always update the GN group-profile with information from the OIDC roles.
+     */
+    public boolean userGroupUpdateEnabled = true;
+
+
+    public String clientId;
+    public String clientSecret;
+    public String scopes = null; //null or empty -> all
+    public String roleConverterString = null;
+
+    public String getScopes() {
+        return scopes;
+    }
+
+    public void setScopes(String scopes) {
+        this.scopes = scopes;
+    }
+
+    public List<String> getScopeSet() {
+        if (!StringUtils.hasText(scopes))
+            return null;
+        return Arrays.asList(scopes.split(" "));
+    }
+
+    public String getUserNameAttribute() {
+        return userNameAttribute;
+    }
+
+    public void setUserNameAttribute(String userNameAttribute) {
+        this.userNameAttribute = userNameAttribute;
+    }
+
+    public String getOrganizationProperty() {
+        return organizationProperty;
+    }
+
+    public void setOrganizationProperty(String organizationProperty) {
+        this.organizationProperty = organizationProperty;
+    }
+
+    public String getGroupPermissionSeparator() {
+        return groupPermissionSeparator;
+    }
+
+    public void setGroupPermissionSeparator(String groupPermissionSeparator) {
+        this.groupPermissionSeparator = groupPermissionSeparator;
+    }
+
+    public String getIdTokenRoleLocation() {
+        return idTokenRoleLocation;
+    }
+
+    public void setIdTokenRoleLocation(String idTokenRoleLocation) {
+        this.idTokenRoleLocation = idTokenRoleLocation;
+    }
+
+    public Map<String, String> getRoleConverter() {
+        return roleConverter;
+    }
+
+    public void setRoleConverter(Map<String, String> roleConverter) {
+        this.roleConverter = roleConverter;
+    }
+
+    public Profile getMinimumProfile() {
+        return Profile.findProfileIgnoreCase(minimumProfile);
+    }
+
+    public void setMinimumProfile(String minimumProfile) {
+        this.minimumProfile = minimumProfile;
+    }
+
+    @Override
+    public String getLoginType() {
+        return LoginType.LINK.toString();
```

Comment:
Is there a reason the AUTOLOGIN option was not added?
In our case, we don't want anonymous user access on our system so we use the AUTOLOGIN so that the user is automatically redirected to the login if they are not logged in.

Reply:
I added the ability to override this in the -overrides file with the environment variable "OPENIDCONNECT_LOGINTYPE".

Defaults to LINK

Reply:
pushed changes to code and doc